.github/workflows/api.yml

```diff
@@ -29,7 +29,24 @@ jobs:
       - name: Run build
         run: make build
 
-      - uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      - name: Checkout developer documentation repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          name: mattermost-api-reference
-          path: api/v4/html
+          repository: mattermost/mattermost-developer-documentation
+          token: ${{ secrets.DEV_DOCS_TOKEN }}
+          path: developer-docs
+
+      - name: Copy OpenAPI spec
+        run: |
+          mkdir -p ${{ github.workspace }}/developer-docs/site/static
+          cp v4/html/static/mattermost-openapi-v4.yaml ${{ github.workspace }}/developer-docs/site/static/
+
+      - name: Commit and push changes
+        if: github.ref == 'refs/heads/master'
+        run: |
+          cd ${{ github.workspace }}/developer-docs
+          git config --global user.name "API Documentation Bot"
+          git config --global user.email "apidocsbot@mattermost.com"
+          git add site/static/mattermost-openapi-v4.yaml
+          git commit -m "Update OpenAPI spec" || echo "No changes to commit"
+          git push
```

Comment:
@toninis Can you have a look at this and let me know if it's the best way to do what we're aiming for? We're building the openAPI yaml file here and we need to push it to the `site/static` folder of mattermost-developer-documentation

Reply:
Hey @esarafianou we have an [automation](https://github.com/mattermost/delivery-platform/blob/master/automations/mattermost/api-reference.yaml) triggered privately to update the API docs [here](https://github.com/mattermost/delivery-platform/blob/master/.github/workflows/release-mattermost-api-reference.yaml) . 
Since we have everything in place there you can reuse also our bot to commit after the release 😄 
I suggest you migrate this process there to avoid hardcoding long lived tokens to our public repo.